apps/web/src/components/Invoices/Table.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  useCancelInvoiceById,
+  useFindInvoices,
+} from "@litespace/headless/invoices";
+import React, { useCallback, useEffect, useState } from "react";
+import { InvoicesTable } from "@litespace/ui/InvoicesTable";
+import { Typography } from "@litespace/ui/Typography";
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import { IInvoice, IWithdrawMethod } from "@litespace/types";
+import { useToast } from "@litespace/ui/Toast";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import { QueryKey } from "@litespace/headless/constants";
+
+function asInvoices({ list }: { list: IInvoice.Self[] }): Array<{
+  id: number;
+  createdAt: string;
+  amount: number;
+  method: IWithdrawMethod.Type;
+  receiver: string;
+  status: Exclude<IInvoice.Status, IInvoice.Status.UpdatedByReceiver>;
+}> {
+  return list
+    .filter((invoice) => invoice.status !== IInvoice.Status.UpdatedByReceiver)
+    .map((invoice) => {
+      return {
+        id: invoice.id,
+        createdAt: invoice.createdAt,
+        amount: invoice.amount,
+        method: invoice.method,
+        receiver: invoice.receiver,
+        status: invoice.status as Exclude<
+          IInvoice.Status,
+          IInvoice.Status.UpdatedByReceiver
+        >,
+      };
+    });
+}
+
+export const Table: React.FC = () => {
+  const intl = useFormatMessage();
+  const toast = useToast();
+  const [open, setOpen] = useState(false); // For delete dialog
+
+  const [id, setId] = useState<number>(0); // The invoice id that will be deleted
+  const invoicesQuery = useFindInvoices({ userOnly: true });
+  const invalidateQuery = useInvalidateQuery();
+
+  const onSuccess = useCallback(() => {
+    invalidateQuery([QueryKey.FindInvoices]);
+    setOpen(false);
+    toast.success({ title: intl("invoices.cancel-toast.success") });
+  }, [invalidateQuery, toast, intl]);
+
+  const onError = useCallback(() => {
+    setOpen(false);
+    toast.error({ title: intl("invoices.cancel-toast.error") });
+  }, [toast, intl]);
+
+  const cancelMutation = useCancelInvoiceById({ id, onSuccess, onError });
+
+  useEffect(() => {
+    if (id === 0) return;
+    cancelMutation.mutate();
+
+    return () => setId(0);
+  }, [cancelMutation, id]);
+
+  if (!invoicesQuery.query.data) return null;
```

Comment:
This is not a valid logic. `query.data` will be null during the loading and errors!